src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Security/AuthorizationHandlers/DeleteSpecificMethodologyAuthorizationHandlerTests.cs

```diff
@@ -135,17 +153,90 @@ public async Task UserWithCorrectClaimCannotDeleteApprovedMethodology()
                         ) = CreateHandlerAndDependencies();
 
                     methodologyVersionRepository
-                        .Setup(s => s.IsPubliclyAccessible(ApprovedMethodologyVersion.Id))
+                        .Setup(s => s.IsPubliclyAccessible(ApprovedFirstVersion.Id))
                         .ReturnsAsync(false);
 
                     var user = CreateClaimsPrincipal(UserId, claim);
-                    var authContext = CreateAuthContext(user, ApprovedMethodologyVersion);
+                    var authContext = CreateAuthContext(user, ApprovedFirstVersion);
 
                     await handler.HandleAsync(authContext);
                     VerifyAllMocks(methodologyVersionRepository, publicationRoleRepository);
 
                     // Verify that the presence of the "DeleteAllMethodologies" Claim still doesn't allow the user to
-                    // do this with a Methodology in this state.
+                    // do this with an approved version.
+                    Assert.False(authContext.HasSucceeded);
+                });
+            }
+
+            [Fact]
+            public async Task UserWithCorrectClaimCanDeleteDraftMethodologyAmendment()
+            {
+                await ForEachSecurityClaimAsync(async claim =>
+                {
+                    var expectClaimToSucceed = claim == DeleteAllMethodologies;
+
+                    await using var context = InMemoryApplicationDbContext(Guid.NewGuid().ToString());
+                    context.Attach(DraftAmendmentVersion);
```

Comment:
Short answer: I've removed it now along with the database context. The handler doesn't rely on the database for anything and all the services are mocked. Nothing needs to be saved.

Explanation on why they were there since that hasn't always been the case and this hander use to depend directly on the database:

There was a change to the handler back on 11th aug which removed this from the body of the `HandleRequirementAsync`

```
            await _context
                .Entry(methodology)
                .Reference(p => p.MethodologyParent)
                .LoadAsync();

            await _context
                .Entry(methodology.MethodologyParent)
                .Collection(p => p.Publications)
                .LoadAsync();
```

Prior to removing this, both these statements would fail if the methodology hasn't been loaded into the database context yet or was somehow known by the context but detached. You can't call `Load`  on properties when the database context isn't tracking the entity.

The security handlers receive entities from the `AuthorizationHandlerContext` which is setup when a service makes a security check. The service making the security check is responsible for loading the entity. The security handler has no control over the entity so can't guarantee the entity is already loaded by the database context.

An easy way to see this code fail would be to query the entity from the database in the service prior to making the security check with `.AsNoTracking`. The entity won't be  loaded in the database context and calling `Load` on a property of the entity within the security handler will fail.

From a testing point of view and the reason for the `Attach`:
The handler was set up with a database context. The entity was attached to the context without being saved. It could also have been saved separately but if this was tested with a fresh context it would still have needed to be attached since the security handler wasn't responsible for loading the entity as discussed.

As an aside, to guarantee that an entity is loaded in a security handler we recently added an extension method `TryReloadEntity` which can be used. See this example in `ViewReleaseAuthorizationHandler`:

```
           if (!_context.TryReloadEntity(release, out var loadedRelease))
            {
                return;
            }

            await _context.Entry(loadedRelease)
                .Reference(p => p.Publication)
                .Query()
                .Include(p => p.Releases)
                .LoadAsync();
```